app/routes.py

```diff
@@ -1 +1,206 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+
+from flask import Blueprint, jsonify, abort, make_response, request
+import datetime
+import os
+import requests
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@task_bp.route("", methods=["POST"])
+def make_a_task():
+    try:
+        request_body = request.get_json()
+
+        new_task = Task.from_dict(request_body)
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return {
+        "task": new_task.to_dict()}, 201
+    except KeyError as error:
+        return {"details": "Invalid data"}, 400
```

Comment:
Love how you implemented a `try/except` block! Like we validated models, could we also make a general function that validates request bodies? Here's an example:
```python
def validate_request_body(request_body, keys):
    for key in keys:
        if not request_body.get(key): 
            abort(make_response({
                'Invalid Data': f'missing key: {key}'
            }, 400))

    return True
```
We can pass in the `request_body` and a list of strings that are `keys` and then check to see if those keys are present.